### DISCUSSION TITLE
Use of calculated kernel values in calculation of functions

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#18455] [Q&A General] [answered]
Is it possible to use a function to calculate a value based on a value called into the function from a kernel?  Is it then possible to set that function value equal to the coef/coefficient of CoupledForce and/or CoefReaction respectively?

Thanks

**Comment by GiudGiud** [comment, marked as answer]
"function to calculate a value based on a value called into the function from a kernel"
well this exactly, not with the Function system. 

However, with the Materials system, you could make these two functions material properties, and one can depend on the other, then be called in MatReaction (the equivalent of CoefReaction but with a material property coefficient). 

It also could be that you can de-convolute this requirement, and do all this computing in a custom kernel. CoefReaction and CoupledForce are very simple kernels, you could very easily modify them.
